motionplan/constraint.go

```diff
@@ -231,64 +230,21 @@ func (c *ConstraintHandler) SegmentConstraints() []string {
 }
 
 func createAllCollisionConstraints(
-	frame *solverFrame,
-	fs referenceframe.FrameSystem,
-	worldState *referenceframe.WorldState,
-	inputs map[string][]referenceframe.Input,
-	pbConstraint []*pb.CollisionSpecification,
+	movingRobotGeometries, staticRobotGeometries, worldGeometries []spatial.Geometry,
+	allowedCollisions []*Collision,
 	collisionBufferMM float64,
 ) (map[string]StateConstraint, error) {
 	constraintMap := map[string]StateConstraint{}
+	var err error
 
-	// extract inputs corresponding to the frame
-	frameInputs, err := frame.mapToSlice(inputs)
-	if err != nil {
-		return nil, err
-	}
-
-	// create robot collision entities
-	movingGeometries, err := frame.Geometries(frameInputs)
-	if err != nil {
-		if len(movingGeometries.Geometries()) == 0 {
-			return nil, err // no geometries defined for frame
-		}
-	}
-
-	// find all geometries that are not moving but are in the frame system
-	staticGeometries := make([]spatial.Geometry, 0)
-	frameSystemGeometries, err := referenceframe.FrameSystemGeometries(fs, inputs)
-	if err != nil {
-		return nil, err
-	}
-	for name, geometries := range frameSystemGeometries {
-		if !frame.movingFrame(name) {
-			staticGeometries = append(staticGeometries, geometries.Geometries()...)
-		}
-	}
-
-	// Note that all obstacles in worldState are assumed to be static so it is ok to transform them into the world frame
-	// TODO(rb) it is bad practice to assume that the current inputs of the robot correspond to the passed in world state
-	// the state that observed the worldState should ultimately be included as part of the worldState message
-	obstacles, err := worldState.ObstaclesInWorldFrame(fs, inputs)
-	if err != nil {
-		return nil, err
-	}
-
-	allowedCollisions, err := collisionSpecificationsFromProto(pbConstraint, frameSystemGeometries, worldState)
-	if err != nil {
-		return nil, err
-	}
-
-	if len(obstacles.Geometries()) > 0 {
```

Comment:
Why is this whole section being moved outside this function?  I might be missing something but it looks like its not being called any differently and this has a consequence of making tests a lot more verbose

Reply:
See other comment